node/network/bridge/src/lib.rs

```diff
@@ -519,6 +538,8 @@ where
 							"Received a validator connection request",
 						);
 
+						metrics.note_desired_peer_count(peer_set, validator_ids.len());
```

Comment:
That does not look quite right. That's only registering a single request, not the total number.

We should be tracking calls to [`add_to_peers_set`](https://github.com/paritytech/polkadot/blob/master/node/network/bridge/src/network.rs#L120) and [`remove_from_peers_set`](https://github.com/paritytech/polkadot/blob/master/node/network/bridge/src/network.rs#L122), to get the correct state.

Reply:
Maybe I'm misunderstanding the code, but I thought that we passed to `on_request` the list of all validators we should be connected to, and then `on_request` updates the actual list of reserved peers to match what was passed as parameter.

Reply:
No, it is additional peers. The connections are kept alive as long as the passed `oneshot` is kept alive.

Reply:
They requests might be overlapping though. So we might count a peer twice in two `add_to_peers_set` calls.
For validation PeerSet we issue two requests:
* to the authorities in the next, current and some previous sessions: 
https://github.com/paritytech/polkadot/blob/cbf2f06627e4e56b85b3951cf8054c188f12349a/node/network/gossip-support/src/lib.rs#L161
* to a subset of that in availability-distribution:
https://github.com/paritytech/polkadot/blob/cbf2f06627e4e56b85b3951cf8054c188f12349a/node/network/availability-distribution/src/pov_requester/mod.rs#L197

we should probably remove the second request, and then this will be correct
however, for the collation PeerSet, we issue two separate requests
* https://github.com/paritytech/polkadot/blob/cbf2f06627e4e56b85b3951cf8054c188f12349a/node/network/collator-protocol/src/collator_side.rs#L342-L351

we could merge them into one maybe? and then we could probably remove the reference counting assuming only one subsystem issues the requests and only needs one request at a time (and also use `set_peers_set` instead of `add_` and `remove_`)

Reply:
yeah, those separate requests are there so we can drop connections individually.